Added looping functionality

Leone25 opened the pull request:
cool shit, ik

Leone25 committed:
Added looping functionality

cool shit, ik

Leone25 committed:
Delete command

Yep, even more cool shit

Leone25 committed:
added last option to delete command

zacwm commented on src/commands/stop.js:
```diff
@@ -3,6 +3,7 @@ const sm = require("../index");
 
 sm.command(["stop", "disconnect", "leave"], (msg) => {
     let cmdGuild = sm.data[msg.guild.id];
+    sm.data[msg.guild.id].loop = false;
```
This isn't required as it is deleted in the disconnect listener to reset it for the next join event.
https://github.com/zacimac/SimpleMusic/blob/f7c7d6ea70286439ccaa793c04d7dcba96b1bdf4/src/commands/events.js#L40

zacwm commented on src/commands/delete.js:
```diff
@@ -0,0 +1,14 @@
+        sm.data[msg.guild.id].queue.splice(sm.data[msg.guild.id].queue.lenght - 1, 1);
+        msg.react("🗑");
+    } else if (!isInteger(msg.args[1]) || msg.args[1] <= 0 || msg.args[1] >= sm.data[msg.guild.id].queue.lenght) {
+        msg.react("❌");
```
Maybe a message indicating the issue instead of a reaction as it could be confused with the reaction for the invalid permissions (role requirement).

zacwm commented on src/commands/delete.js:
```diff
@@ -0,0 +1,14 @@
+
+sm.command(["delete", "d"], async (msg) => {
+    if (msg.args[1] == "last") {
+        sm.data[msg.guild.id].queue.splice(sm.data[msg.guild.id].queue.lenght - 1, 1);
```
`.length` is spelt incorrectly for lines 6 & 8

Leone25 committed:
Fixing . . .

Leone25 commented:
Dam it!

Hopefully good now

Leone25 committed:
Changes to prepare support for multiple sources

Leone25 committed:
woops, missed one thing

Leone25 committed:
Added soundcloud support

I'm nto too sure if we want to keep both `ytdl-core` and `youtube-dl` tho if we want we can easily switch to only use `youtube-dl`, i'll let you decide

Leone25 commented:
so it's apparently merging the 2 modifications in to a single pullup request, tho i'm added soundcloud support
I'm not too sure if we want to keep both `ytdl-core` and `youtube-dl` tho if we want we can easily switch to only use `youtube-dl`, i'll let you decide

zacwm committed:
Number.isInteger() fix

zacwm committed:
Looping moved to playSong() function

If looping was enabled either during a single playing song or standby, the next queued song will not be looped.
By moving it to the playSong() function, it will immediately requeue the song when it starts playing, enabling the loop to work better.
It is also possible to have detection in the `loop.js` command to detect if a song is playing to queue the playing song too, but maybe?? ;)

zacwm committed:
Cleanup and colour fix.

The warn and error shouldn't be used for successful setting changes.
